accumulate.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudscale-ch/cloudscale-go-sdk/v2"
+	"os"
+	"time"
+)
+
+type AccumulateKey struct {
+	Query     string
+	Zone      string
+	Tenant    string
+	Namespace string
+	Start     time.Time
+}
+
+func (this AccumulateKey) String() string {
+	return this.Query + ":" + this.Zone + ":" + this.Tenant + ":" + this.Namespace
+}
+
+func accumulate(ctx context.Context, date time.Time, cloudscaleClient *cloudscale.Client) (map[AccumulateKey]uint64, error) {
+	bucketMetricsRequest := cloudscale.BucketMetricsRequest{Start: date, End: date}
+	bucketMetrics, err := cloudscaleClient.Metrics.GetBucketMetrics(ctx, &bucketMetricsRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	accumulated := make(map[AccumulateKey]uint64)
+
+	for _, bucketMetricsData := range bucketMetrics.Data {
+		objectsUser, err := cloudscaleClient.ObjectsUsers.Get(ctx, bucketMetricsData.Subject.ObjectsUserID)
+		if err != nil || objectsUser == nil {
+			fmt.Fprintf(os.Stderr, "WARNING: Cannot sync bucket %s, objects user %s not found\n", bucketMetricsData.Subject.BucketName, bucketMetricsData.Subject.ObjectsUserID)
+			continue
+		}
```

Comment:
The way this function is designed makes it very hard to unit test, since it depends on the cloudscale client. To unit test, we would need to mock/stub it.
However, I believe there's a way where we can at least test this function partially in a unit test.
Lines 39 to 79 can be extracted into its own function (which accepts the metricsData and objectsUser). That way we can test whether the objectsUser have the correct tags, the AccumulateKey has the correct properties and ultimately accumulate the correct numbers.
This new function we can unit test by constructing the objectsUser and Metricsdata by ourselves without making calls to the cloudscale API.

Reply:
Done.